CMakeLists.txt

```diff
@@ -232,3 +232,10 @@ if(WITH_ORACLE)
     add_subdirectory(oracle)
     SET(CPACK_INSTALL_CMAKE_PROJECTS "release/oracle.app;oracle;ALL;/" ${CPACK_INSTALL_CMAKE_PROJECTS})
 endif()
+
+# Compile tests (default off)
+option(WITH_TESTS "build tests" OFF)
+if(WITH_TESTS)
```

Comment:
Afaik no special "test" variable exists in cmake: https://cmake.org/cmake/help/v3.0/manual/cmake-variables.7.html
A special "test" target exists when CTest is enabled, but "-Dtest=ON" is nothing more than a normal #define.
So, tried to make it homogeneous with the other target-enabling definitions  (WITH_ORACLE, WITH_SERVER, ..).